### PR TITLE
Enhancements

### PR DESCRIPTION
**Pull request by flav-code**
Minor improvement

Added Rest.getLavalinkInfo() and his type
> https://github.com/lavalink-devs/Lavalink/blob/master/IMPLEMENTATION.md#get-lavalink-info

### CODE REVIEW

File: src/guild/Connection.ts [hunk 1]
```diff
@@ -226,6 +226,14 @@ export class Connection extends EventEmitter {
         this.debug(`[Voice] <- [Discord] : Server Update Received | Server: ${this.region} Guild: ${this.guildId}`);
     }
 
+    /**
```
Comment (Deivu) [inline src/guild/Connection.ts:229]: I'm not sure about this change, as disconnect() will supply its own method to this.send, instead of having them use this function all together for simplicity

**Review by Deivu — COMMENTED** [review]
Check above for review

**Comment by flav-code** [comment]
well, I made the function to avoid duplicating coe but you can add an argument for when you need to disconnect the bot or set channel_id to null before calling the function.

**Comment by Deivu** [comment]
I just dont really turn it to null before in a case you want to reconnect, but I guess that should be on developer's responsibility eh?

**Comment by flav-code** [comment]
That's why I didn't put the function for disconnect, I only put it where it's the same.

**Comment by flav-code** [comment]
but what are you talking about

**Comment by Deivu** [comment]
Reset every data in the connection class, before calling `this.sendVoiceUpdate()`, so it's up toe the developer's (user) responsibility to cache the existing data

**Comment by flav-code** [comment]
so in disconnect you call the function and you want the user to delete the channel_id before calling the disconnect function?

**Comment by Deivu** [comment]
> so in disconnect you call the function and you want the user to delete the channel_id before calling the disconnect function?

Tldr: 
```js
this.channel_id = null;
this.deafened = false;
this.muted = false;
this.sendVoiceUpdate();
// ... rest of the code in disconnect
```

**Comment by flav-code** [comment]
That's what I wanted to do...
Is that all right with you?
Shall I add it?

**Comment by Deivu** [comment]
For now yeah, as I stated above.

> Yeah, but I'm thinking if we should let the developer do that instead, after all I think anything outside the scope of lavalink & discord api should be in developer's responsibility, not our responsibility. I'll re-review once decided.

To keep things simple and concise

**Comment by flav-code** [comment]
Okay, but this is just to make the code simpler and it won't change anything for the user.

**Comment by Deivu** [comment]
*probably* kek, if they dont cache channel id and on closed event, they get the channelid from this class, then sure it will break, but then again, im passing the responsibility of caching this to them now

**Comment by flav-code** [comment]
Oh okay, I get it, I'm actually using the Connection.channelId

**Review by Deivu — CHANGES_REQUESTED** [review]
Please do the changes I said above on disconnect to keep things consistent

File: src/guild/Connection.ts [hunk 2]
```diff
@@ -135,10 +135,7 @@ export class Connection extends EventEmitter {
      * @internal
      */
     public disconnect(): void {
-        this.channelId = null;
```
Comment (Deivu) [inline src/guild/Connection.ts]: Oh, I didn't mean to remove this, I didn't notice you actually already did it, just revert this and I'll check again. Thought it was not yet done

**Review by Deivu — CHANGES_REQUESTED** [review]

**Comment by flav-code** [comment]
lol

**Comment by flav-code** [comment]
good

**Review by Deivu — APPROVED** [review]
Looking good
